### ISSUE TITLE
Support for st

### ISSUE THREAD

**Issue by HalosGhost** [stayradiated/terminal.sexy#22]
The [suckless terminal](http://st.suckless.org/)'s config is a C header, and the section of the config that deals with colors is just a simple array. It would be awesome to see support for it added!

**Comment by stayradiated** [comment]
Would something like this be alright?

```
static const char *colorname[] = {
  "#073642",  /*  0 */
  "#dc322f",  /*  1 */
  "#859900",  /*  2 */
  "#b58900",  /*  3 */
  "#268bd2",  /*  4 */
  "#d33682",  /*  5 */
  "#2aa198",  /*  6 */
  "#eee8d5",  /*  7 */
  "#002b36",  /*  8 */
  "#cb4b16",  /*  9 */
  "#586e75",  /* 10 */
  "#657b83",  /* 11 */
  "#839496",  /* 12 */
  "#6c71c4",  /* 13 */
  "#93a1a1",  /* 14 */
  "#fdf6e3",  /* 15 */
};
```

**Comment by HalosGhost** [comment]
Personally, I think it'd be nice if the comments were a little more descriptive (the first sixteen colors are the ANSI color codes, the first eight standard, the second eight “bright” (both in ANSI escape order).

Additionally, there are a few more settings available, ``defaultfg``, ``defaultbg``, ``defaultcs``, ``defaultitalic``, and ``defaultunderline``. These settings are indecies in the ``colorname`` array. While the ``colorname`` array can include 256 colors for standard 256-color, these settings can be set to special colors which are defined as indecies beyond 255. For example, my current ``config.h`` includes the following:

```c
static const char * colorname [] = {
    ...
    [255] = 0,

    /* more colors can be added after 255 to use with DefaultXX */
    "#cccccc",
    "#333333"
};

/*
 * Default colors (colorname index)
 * foreground, background, cursor
 */
static unsigned int defaultfg = 7;
static unsigned int defaultbg = 257;
static unsigned int defaultcs = 256;

/*
 * Colors used, when the specific fg == defaultfg. So in reverse mode this
 * will reverse too. Another logic would only make the simple feature too
 * complex.
 */
static unsigned int defaultitalic = 7;
static unsigned int defaultunderline = 7;
```

**Comment by stayradiated** [comment]
Ok. How about this?

``` c
/* Terminal colors (16 first used in escape sequence) */
static const char *colorname[] = {
  /* 8 normal colors */
  "#073642",  /*  0: black   */
  "#dc322f",  /*  1: red     */
  "#859900",  /*  2: green   */
  "#b58900",  /*  3: yellow  */
  "#268bd2",  /*  4: blue    */
  "#d33682",  /*  5: magenta */
  "#2aa198",  /*  6: cyan    */
  "#eee8d5",  /*  7: white   */

  /* 8 bright colors */
  "#002b36",  /*  8: black   */
  "#cb4b16",  /*  9: red     */
  "#586e75",  /* 10: green   */
  "#657b83",  /* 11: yellow  */
  "#839496",  /* 12: blue    */
  "#6c71c4",  /* 13: magenta */
  "#93a1a1",  /* 14: cyan    */
  "#fdf6e3",  /* 15: white   */

  [255] = 0,

  /* special colors */
  "#002b36",  /* 256: background */
  "#839496",  /* 257: foreground */
};

/*
 * Default colors (colorname index)
 * foreground, background, cursor
 */
static unsigned int defaultfg = 257;
static unsigned int defaultbg = 256;
static unsigned int defaultcs = 257;

/*
 * Colors used, when the specific fg == defaultfg. So in reverse mode this
 * will reverse too. Another logic would only make the simple feature too
 * complex.
 */
static unsigned int defaultitalic = 7;
static unsigned int defaultunderline = 7;
```

**Comment by HalosGhost** [comment]
That is a much more complete case. The only other thing I might recommend (and this is purely cosmetic, so it is entirely up to you) is that you can move the numbers into indecies instead of the comments and that might look cleaner:

```c
static const char * colorname [] = {
    [0] = "#073642", /* black */
    [1] = "#dc322f", /* red */
    …
};

It's the same style that the ``255`` index already uses.

But either way, this looks great!

**Comment by stayradiated** [comment]
Is this syntax correct? If so, I'll go ahead and add it to terminal.sexy.

``` c
/* Terminal colors (16 first used in escape sequence) */
static const char *colorname[] = {

  /* 8 normal colors */
  [0] = "#073642", /* black   */
  [1] = "#dc322f", /* red     */
  [2] = "#859900", /* green   */
  [3] = "#b58900", /* yellow  */
  [4] = "#268bd2", /* blue    */
  [5] = "#d33682", /* magenta */
  [6] = "#2aa198", /* cyan    */
  [7] = "#eee8d5", /* white   */

  /* 8 bright colors */
  [8]  = "#002b36", /* black   */
  [9]  = "#cb4b16", /* red     */
  [10] = "#586e75", /* green   */
  [11] = "#657b83", /* yellow  */
  [12] = "#839496", /* blue    */
  [13] = "#6c71c4", /* magenta */
  [14] = "#93a1a1", /* cyan    */
  [15] = "#fdf6e3", /* white   */

  /* required by st */
  [255] = 0,

  /* special colors */
  [256] = "#002b36", /* background */
  [257] = "#839496", /* foreground */
};

/*
 * Default colors (colorname index)
 * foreground, background, cursor
 */
static unsigned int defaultfg = 257;
static unsigned int defaultbg = 256;
static unsigned int defaultcs = 257;

/*
 * Colors used, when the specific fg == defaultfg. So in reverse mode this
 * will reverse too. Another logic would only make the simple feature too
 * complex.
 */
static unsigned int defaultitalic = 7;
static unsigned int defaultunderline = 7;
```

**Comment by HalosGhost** [comment]
That looks wonderful!

**Comment by stayradiated** [comment]
I've added it to the [termcolors](https://github.com/stayradiated/termcolors) library that is used to import & export schemes. 

I'll push the update to terminal.sexy within the next 24 hours.

**Comment by stayradiated** [comment]
Added in v0.2.7 - which has just been deployed.

**Comment by HalosGhost** [comment]
It seems like this was removed. Despite being present in the termcolors library, it doesn't look like terminal.sexy supports exporting to it. Did something go wrong?

**Comment by stayradiated** [comment]
The v0.2.7 accidentally broke other parts of the app, so I had to roll it back to an earlier version.
See https://github.com/stayradiated/terminal.sexy/issues/23 for the original issue.

**Comment by HalosGhost** [comment]
Ahh, I see. Thank you for keeping the small-time-terminal users in-mind!

**Comment by stayradiated** [comment]
I've fixed the issue with v0.2.7 so you can export as `st` again.

**Comment by ubone** [comment]
there's an issue with this as of writing this comment
```
static unsigned int defaultfg = 257;
static unsigned int defaultbg = 256;
```
should be
```
unsigned int defaultfg = 257;
unsigned int defaultbg = 256;
```
without those changes compile fails with:
```
In file included from x.c:61:
config.h:117:21: error: static declaration of ‘defaultfg’ follows non-static declaration
 static unsigned int defaultfg = 257;
                     ^~~~~~~~~
In file included from x.c:20:
st.h:122:21: note: previous declaration of ‘defaultfg’ was here
 extern unsigned int defaultfg;
                     ^~~~~~~~~
In file included from x.c:61:
config.h:118:21: error: static declaration of ‘defaultbg’ follows non-static declaration
 static unsigned int defaultbg = 256;
                     ^~~~~~~~~
In file included from x.c:20:
st.h:123:21: note: previous declaration of ‘defaultbg’ was here
 extern unsigned int defaultbg;
                     ^~~~~~~~~
make: *** [Makefile:22: x.o] Error 1
make: *** Waiting for unfinished jobs....
==> ERROR: A failure occurred in build().
    Aborting...
```
